src/test/java/org/cryptomator/cryptofs/attr/CryptoDosFileAttributesTest.java

```diff
@@ -43,77 +43,85 @@ public void setup() {
+			Assertions.assertEquals(value, inTest.isArchive());
+			verify(delegate, times(1)).isArchive();
 		}
 
 		@DisplayName("isHidden()")
 		@ParameterizedTest(name = "is {0} if delegate.isHidden() is {0}")
 		@CsvSource({"true", "false"})
-		public void testIsHiddenDelegates(boolean value) {
+		public void testIsHiddenImmutable(boolean value) {
 			when(delegate.isHidden()).thenReturn(value);
+			inTest = new CryptoDosFileAttributes(delegate, FILE, path, cryptor, Optional.of(openCryptoFile), cryptoFileSystemProperties);
 
-			Assertions.assertSame(value, inTest.isHidden());
+			verify(delegate, times(1)).isHidden();
+			Assertions.assertEquals(value, inTest.isHidden());
+			verify(delegate, times(1)).isHidden();
 		}
 
 		@DisplayName("isReadOnly()")
 		@ParameterizedTest(name = "is {0} if delegate.readOnly() is {0}")
 		@CsvSource({"true", "false"})
-		public void testIsReadOnlyDelegates(boolean value) {
+		public void testIsReadOnlyImmutable(boolean value) {
 			when(delegate.isReadOnly()).thenReturn(value);
+			inTest = new CryptoDosFileAttributes(delegate, FILE, path, cryptor, Optional.of(openCryptoFile), cryptoFileSystemProperties);
 
-			Assertions.assertSame(value, inTest.isReadOnly());
+			verify(delegate, times(1)).isReadOnly();
+			Assertions.assertEquals(value, inTest.isReadOnly());
+			verify(delegate, times(1)).isReadOnly();
 		}
 
 		@DisplayName("isSystem()")
 		@ParameterizedTest(name = "is {0} if delegate.isSystem() is {0}")
 		@CsvSource({"true", "false"})
-		public void testIsSystemDelegates(boolean value) {
+		public void testIsSystemImmutable(boolean value) {
 			when(delegate.isSystem()).thenReturn(value);
+			inTest = new CryptoDosFileAttributes(delegate, FILE, path, cryptor, Optional.of(openCryptoFile), cryptoFileSystemProperties);
 
-			Assertions.assertSame(value, inTest.isSystem());
+			verify(delegate, times(1)).isSystem();
+			Assertions.assertEquals(value, inTest.isSystem());
+			verify(delegate, times(1)).isSystem();
 		}
 
 	}
 
 	@Nested
-	@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 	@DisplayName("on read-only filesystem")
 	public class ReadOnlyFileSystem {
 
-		private CryptoDosFileAttributes inTest;
-
-		@BeforeAll
-		public void setup() {
+		@BeforeEach
+		public void beforeEach() {
 			when(cryptoFileSystemProperties.readonly()).thenReturn(true);
-			inTest = new CryptoDosFileAttributes(delegate, FILE, path, cryptor, Optional.of(openCryptoFile), cryptoFileSystemProperties);
 		}
 
 		@DisplayName("isReadOnly()")
 		@ParameterizedTest(name = "is true if delegate.readOnly() is {0}")
 		@CsvSource({"true", "false"})
 		public void testIsReadOnlyForReadonlyFileSystem(boolean value) {
 			when(delegate.isReadOnly()).thenReturn(value);
+			var inTest = new CryptoDosFileAttributes(delegate, FILE, path, cryptor, Optional.of(openCryptoFile), cryptoFileSystemProperties);
 
+			verify(delegate, times(1)).isReadOnly();
 			Assertions.assertTrue(inTest.isReadOnly());
+			verify(delegate, times(1)).isReadOnly();
```

Comment:
`isReadOnly()` isn't necessarily invoked:

https://github.com/cryptomator/cryptofs/blob/59b4ce885356092922db28d70b5e6ee657dcb636/src/main/java/org/cryptomator/cryptofs/attr/CryptoDosFileAttributes.java#L34

```suggestion
			verify(delegate, Mockito.atMostOnce()).isReadOnly();
			Assertions.assertTrue(inTest.isReadOnly());
			verify(delegate, Mockito.atMostOnce()).isReadOnly();
```